controllers/syncer/csi_node.go

```diff
@@ -342,42 +339,31 @@ func (s *csiNodeSyncer) ensureVolumes() []corev1.Volume {
 	}
 }
 
-func (s *csiNodeSyncer) getSidecarByName(name string) *csiv1.CSISidecar {
-	return getSidecarByName(s.driver, name)
-}
-
-func (s *csiNodeSyncer) getCSINodeDriverRegistrarImage() string {
-	sidecar := s.getSidecarByName(config.CSINodeDriverRegistrar)
-	if sidecar != nil {
-		return fmt.Sprintf("%s:%s", sidecar.Repository, sidecar.Tag)
-	}
-	return "us.gcr.io/k8s-artifacts-prod/sig-storage/csi-node-driver-registrar:v2.3.0"
-	//return s.driver.GetDefaultSidecarImageByName(config.CSINodeDriverRegistrar)
-}
-
-func (s *csiNodeSyncer) getLivenessProbeImage() string {
-	sidecar := s.getSidecarByName(config.LivenessProbe)
-	if sidecar != nil {
-		return fmt.Sprintf("%s:%s", sidecar.Repository, sidecar.Tag)
-	}
-	return "us.gcr.io/k8s-artifacts-prod/sig-storage/livenessprobe:v2.4.0"
-	//return s.driver.GetDefaultSidecarImageByName(config.LivenessProbe)
-}
-
-func (s *csiNodeSyncer) getCSINodeDriverRegistrarPullPolicy() corev1.PullPolicy {
-	sidecar := s.getSidecarByName(config.CSINodeDriverRegistrar)
-	if sidecar != nil && sidecar.ImagePullPolicy != "" {
-		return sidecar.ImagePullPolicy
-	}
-	return corev1.PullIfNotPresent
-}
+// getImage gets and returns the images for CSI driver from CR
+// if defined in CR, otherwise returns the default images.
+func (s *csiNodeSyncer) getImage(name string) string {
+	logger := csiLog.WithName("getImage")
+	logger.Info("in getImage", "name", name)
 
-func (s *csiNodeSyncer) getLivenessProbePullPolicy() corev1.PullPolicy {
-	sidecar := s.getSidecarByName(config.LivenessProbe)
-	if sidecar != nil && sidecar.ImagePullPolicy != "" {
-		return sidecar.ImagePullPolicy
+	image := ""
+	switch name {
+	case config.CSINodeDriverRegistrar:
+		if len(s.driver.Spec.DriverRegistrar) != 0 {
+			image = s.driver.Spec.DriverRegistrar
+		} else {
+			image = s.driver.GetDefaultImage(name)
+		}
+	case config.GetNameForResource(config.CSINode, s.driver.Name):
```

Comment:
Can we simplify this ? Call csiNodeDriver := config.GetNameForResource(config.CSINode, s.driver.Name) before the switch case and then use that here as "case csiNodeDriver"

Reply:
Done.